Scripts/Core/Contexts/EditingContextManager.cs

```diff
@@ -0,0 +1,282 @@
+﻿#if UNITY_EDITOR && UNITY_EDITORVR
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.InputNew;
+using System.IO;
+
+namespace UnityEditor.Experimental.EditorVR.Core
+{
+	class EditingContextManager : MonoBehaviour
+	{
+		const string k_SettingsPath = "ProjectSettings/EditingContextManagerSettings.asset";
+		const string k_UserSettingsPath = "Library/EditingContextManagerSettings.asset";
+
+		static EditingContextManager s_Instance;
+		static InputManager s_InputManager;
+
+		EditingContextManagerSettings m_Settings;
+
+		List<IEditingContext> m_AvailableContexts;
+		string[] m_ContextNames;
+		int m_SelectedContextIndex;
+
+		IEditingContext m_CurrentContext;
+		readonly List<IEditingContext> m_PreviousContexts = new List<IEditingContext>();
+
+		IEditingContext defaultContext
+		{
+			get
+			{
+				var context = m_AvailableContexts.First();
```

Comment:
I've addressed your point of having a default context. This would also allow us to have a Tutorial context when that time comes.